sky/skylet/constants.py

```diff
@@ -121,12 +123,18 @@
     # latest ray port 6380, but those existing cluster launched before #1790
     # that has ray cluster on the default port 6379 will be upgraded and
     # restarted.
+    'echo PATH=$PATH; '
     f'{SKY_PIP_CMD} list | grep "ray " | '
     f'grep {SKY_REMOTE_RAY_VERSION} 2>&1 > /dev/null '
     f'|| {RAY_STATUS} || '
-    f'{SKY_PIP_CMD} install --exists-action w -U ray[default]=={SKY_REMOTE_RAY_VERSION};'  # pylint: disable=line-too-long
+    f'{SKY_PIP_CMD} install --exists-action w -U ray[default]=={SKY_REMOTE_RAY_VERSION}; '  # pylint: disable=line-too-long
+    # Add missing PATH to make sure ray is in the PATH, when the
+    # previous ray installation happens in user's `~/.local` directory.
+    # ~/.local/bin is added to the end of PATH to avoid conflicts with ray just
+    # installed in the conda environment.
+    'export PATH=$PATH:$HOME/.local/bin; '
```

Comment:
Do I understand correctly:
- The issue is in some envs, our own Ray is by default installed into ~/.local/bin 
- In that env, this path is not in $PATH by default
- So, previously we touch the SKY_RAY_PATH_FILE file, which is empty
- Bug is, later on when we do `our_ray start`, `our_ray` is basically empty string?

Reply:
Yes, this is exactly the issue. Tried to elaborate the comments a bit. PTAL : )